[Flaky] test_softmax_with_length asserts equality near zero

```
======================================================================

FAIL: test_operator.test_softmax_with_length

----------------------------------------------------------------------

Traceback (most recent call last):

  File "C:\Python37\lib\site-packages\nose\case.py", line 198, in runTest

    self.test(*self.arg)

  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\common.py", line 177, in test_new

    orig_test(*args, **kwargs)

  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\test_operator.py", line 5349, in test_softmax_with_length

    [np.zeros(shape), np.zeros(len_shape, dtype=np.int32)], rtol=1e-2, atol=1e-3, dtype="asnumpy")

  File "C:\jenkins_slave\workspace\ut-python-gpu\windows_package\python\mxnet\test_utils.py", line 1228, in check_symbolic_backward

    equal_nan=equal_nan)

  File "C:\jenkins_slave\workspace\ut-python-gpu\windows_package\python\mxnet\test_utils.py", line 533, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 1.240410 exceeds tolerance rtol=0.010000, atol=0.001000.  Location of maximum error:(7, 2, 6), a=0.000000, b=0.001256

 EXPECTED_data: array([[[0., 0., 0., ..., 0., 0., 0.],

        [0., 0., 0., ..., 0., 0., 0.],

        [0., 0., 0., ..., 0., 0., 0.],...

 BACKWARD_data: array([[[-0.00000697, -0.00003457, -0.00001687, ...,  0.        ,

         -0.00000459, -0.00008488],

        [-0.00010872, -0.00010991, -0.00003153, ...,  0.        ,...

-------------------- >> begin captured logging << --------------------
```
[softmax_flaky.log](https://github.com/apache/incubator-mxnet/files/3643365/softmax_flaky.log)

--- a/tests/python/unittest/test_operator.py
+++ b/tests/python/unittest/test_operator.py
@@ -5419,7 +5419,8 @@ def test_softmax_with_length():
         atol = 1e-4 if dtype == np.float16 else 1e-5
         check_symbolic_forward(mx_sym, location, [np_out], rtol=rtol, atol=atol, dtype="asnumpy")
         check_symbolic_backward(mx_sym, location, [np.ones(shape, dtype=dtype)],
-                                [np.zeros(shape), np.zeros(len_shape, dtype=np.int32)], rtol=1e-2, atol=1e-3, dtype="asnumpy")
+                                [np.zeros(shape), np.zeros(len_shape, dtype=np.int32)],
+                                rtol=1e-2, atol=2e-3 if dtype == np.float16 else 1e-3, dtype="asnumpy")
 
 
 @with_seed()

